build-system/babel-plugins/babel-plugin-transform-default-assignment/test/fixtures/transform-assertions/should-transform-destructures/output.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+function test() {
+  let _a = a,
+      _bb = bb,
+      _c = c;
+  const {
+    a = 1,
+    b: bb = 2
+  } = param;
+  const [c = 3] = array;
+  _a;
```

Comment:
Further, assuming the original code is like this:

```
const {a = 1} = param;
a = 2; // That would normally be an error due to const reassignment.
```

And transformed code is like this:
```
const {a = 1} = param;
let _a = a;
_a = 2; // Technically ok now.
```

Could this cause false negatives for us for const assignments? Or is there a check before we get this far?

Reply:
The linter will catch these for us. By the time we're transforming, we're assuming code is correct.